File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.cs
public class Solution {
    public int MaxSum(int[] nums) {
        HashSet<int> unique = new HashSet<int>();
        int sum = 0;
        foreach(int num in nums){
            if(num < 0 || unique.Contains(num))continue;
            unique.Add(num);
            sum+=num;
        }
        if(unique.Count == 0)
            return nums.Max();
        return sum;
    }
}

File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.cpp
class Solution {
public:
    int maxSum(vector<int>& nums) {
        unordered_set<int> unique;
        int sum = 0;
        for(auto &num:nums){
            if(unique.count(num) || num < 0)continue;
            unique.insert(num);
            sum+=num;
        }
        if(unique.size() == 0)
            return *max_element(nums.begin(),nums.end());  
        return sum;
    }
};